test/compiler/AbstractInterpreter.jl

```diff
@@ -514,3 +514,20 @@ let src = code_typed((Int,); interp=CustomDataInterp()) do x
     @test count(isinvoke(:cos), src.code) == 1
     @test count(isinvoke(:+), src.code) == 0
 end
+
+# external cache mode
+@newinterp ExtCacheInterp #=cache_externally=#true
```

Comment:
The phrase "externally" doesn't quite convey the meaning.

Maybe ephemeral or "session" cache. The default cache is an integrated (potentially) persistent cache (even internal is a bit weird, what is it internal to?) 

Whereas this is a ephermal non-integrated cache.
The docs should also probably say that this cache will not be invalidated and should only be used for development